File: symfony-app/src/Entity/UserScoreHistory.php
<?php

namespace App\Entity;

use App\Doctrine\Id\ManualIntId;
use App\Enum\UserScoreSnapshotType;
use App\Repository\UserScoreHistoryRepository;
use Doctrine\ORM\Mapping as ORM;

#[ORM\Entity(repositoryClass: UserScoreHistoryRepository::class)]
#[ORM\Table(name: 'user_score_history')]
#[ORM\Index(name: 'idx_user_score_history_user_created', columns: ['user_id', 'created_at'])]
class UserScoreHistory
{
    #[ORM\Id]
    #[ORM\Column(type: 'integer')]
    private ?int $id = null;

    #[ORM\ManyToOne(targetEntity: User::class)]
    #[ORM\JoinColumn(nullable: false, onDelete: 'CASCADE')]
    private User $user;

    #[ORM\Column(type: 'smallint')]
    private int $score = 0;

    #[ORM\Column(type: 'smallint')]
    private int $activityScore = 0;

    #[ORM\Column(type: 'smallint')]
    private int $seniorityScore = 0;

    #[ORM\Column(type: 'smallint')]
    private int $ruleComplianceScore = 0;

    #[ORM\Column(type: 'smallint')]
    private int $sanctionsHistoryScore = 0;

    #[ORM\Column(type: 'user_score_snapshot_type', length: 50, options: ['default' => 'daily'])]
    private UserScoreSnapshotType $snapshotType = UserScoreSnapshotType::DAILY;

    #[ORM\Column(type: 'datetimetz_immutable')]
    private \DateTimeImmutable $createdAt;

    public function __construct()
    {
        $this->id = ManualIntId::generate();
        $this->createdAt = new \DateTimeImmutable();
    }

    public function getId(): ?int { return $this->id; }

    public function getUser(): User { return $this->user; }
    public function setUser(User $user): void { $this->user = $user; }

    public function getScore(): int { return $this->score; }
    public function setScore(int $score): void { $this->score = $score; }

    public function getActivityScore(): int { return $this->activityScore; }
    public function setActivityScore(int $activityScore): void { $this->activityScore = $activityScore; }

    public function getSeniorityScore(): int { return $this->seniorityScore; }
    public function setSeniorityScore(int $seniorityScore): void { $this->seniorityScore = $seniorityScore; }

    public function getRuleComplianceScore(): int { return $this->ruleComplianceScore; }
    public function setRuleComplianceScore(int $ruleComplianceScore): void { $this->ruleComplianceScore = $ruleComplianceScore; }

    public function getSanctionsHistoryScore(): int { return $this->sanctionsHistoryScore; }
    public function setSanctionsHistoryScore(int $sanctionsHistoryScore): void { $this->sanctionsHistoryScore = $sanctionsHistoryScore; }

    public function getSnapshotType(): UserScoreSnapshotType { return $this->snapshotType; }
    public function setSnapshotType(UserScoreSnapshotType $snapshotType): void { $this->snapshotType = $snapshotType; }

    public function getCreatedAt(): \DateTimeImmutable { return $this->createdAt; }
    public function forceCreatedAt(\DateTimeImmutable $createdAt): void { $this->createdAt = $createdAt; }
}

File: javafx-app/src/main/java/com/santea/model/UserScoreHistory.java
package com.santea.model;

import java.time.LocalDateTime;

public class UserScoreHistory {
    private Integer id;
    private User user;
    private Integer score;
    private Integer activityScore;
    private Integer seniorityScore;
    private Integer ruleComplianceScore;
    private Integer sanctionsHistoryScore;
    private UserScoreSnapshotType snapshotType;
    private LocalDateTime createdAt;

    public UserScoreHistory() {
    }

    public Integer getId() {
        return id;
    }

    public void setId(Integer id) {
        this.id = id;
    }

    public User getUser() {
        return user;
    }

    public void setUser(User user) {
        this.user = user;
    }

    public Integer getScore() {
        return score;
    }

    public void setScore(Integer score) {
        this.score = score;
    }

    public Integer getActivityScore() {
        return activityScore;
    }

    public void setActivityScore(Integer activityScore) {
        this.activityScore = activityScore;
    }

    public Integer getSeniorityScore() {
        return seniorityScore;
    }

    public void setSeniorityScore(Integer seniorityScore) {
        this.seniorityScore = seniorityScore;
    }

    public Integer getRuleComplianceScore() {
        return ruleComplianceScore;
    }

    public void setRuleComplianceScore(Integer ruleComplianceScore) {
        this.ruleComplianceScore = ruleComplianceScore;
    }

    public Integer getSanctionsHistoryScore() {
        return sanctionsHistoryScore;
    }

    public void setSanctionsHistoryScore(Integer sanctionsHistoryScore) {
        this.sanctionsHistoryScore = sanctionsHistoryScore;
    }

    public UserScoreSnapshotType getSnapshotType() {
        return snapshotType;
    }

    public void setSnapshotType(UserScoreSnapshotType snapshotType) {
        this.snapshotType = snapshotType;
    }

    public LocalDateTime getCreatedAt() {
        return createdAt;
    }

    public void setCreatedAt(LocalDateTime createdAt) {
        this.createdAt = createdAt;
    }
}
